### PR TITLE
Integrate with NERC Keycloak

### PR DESCRIPTION
**Pull request by knikolla**
Client definition created in https://github.com/nerc-project/mss-keycloak/blob/main/k8s/overlays/nerc-shift-1/clients/ocp-prod.yaml

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202099997018465

### CODE REVIEW

**Comment by joachimweyl** [comment]
@knikolla I see the other two ColdFront NERC integration PR's are merged. What are the next steps on this PR to get ColdFront on NERC?

File: cluster-scope/overlays/nerc-ocp-prod/oauths/cluster_patch.yaml
Comment (naved001) [inline cluster-scope/overlays/nerc-ocp-prod/oauths/cluster_patch.yaml:17]: just wanted to make sure this is what we are calling it instead of nerc-ocp-prod

Comment (knikolla) [reply cluster-scope/overlays/nerc-ocp-prod/oauths/cluster_patch.yaml:17]: [Correct](https://github.com/nerc-project/mss-keycloak/blob/ca40cfde16e44ecc60a00f369ce9bbaa666448eb/k8s/overlays/nerc-shift-1/clients/ocp-prod.yaml#L37), the client for openstack was just named openstack, and nerc seems reduntant.

**Review by naved001 — APPROVED** [review]

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml
Comment (knikolla) [inline cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml:16]: By looking at the documentation, it seems that when using `patches` you need to specify both `path` and `target`, which wasn't the case here. So I revised this to just be a strategic merge.

File: cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml]: We're using namespaced secret stores on nerc-ocp-prod (because a cluster-wide secret store could unintentially expose secrets in non-admin namespaces). The simplest change is probably to modify `kind:` here to `SecretStore`, and then we'll need another PR creating a `nerc-cluster-secrets` `SecretStore` in the `openshift-config` namespace.

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: You don't need both `path` and `target` for `patches`. You can just write:

```
patches:
- ingresscontrollers/default_patch.yaml
- oauths/cluster_patch.yaml
```

I prefer this (a single `patches` section) over `patchesStrategicmerge` and `patchesJson6902`.

This is equivalent to the above, but arguably more correct:

```
patches:
- path: ingresscontrollers/default_patch.yaml
- path: oauths/cluster_patch.yaml
```

You only need to provide a `target:` when using JSONpatches (because they don't contain any metadata identifying to which resources they should apply).

**Review by larsks — CHANGES_REQUESTED** [review]

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml
Comment (knikolla) [reply cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: Interesting. Will update. GoLand (IDE) was flagging it as an error and I only did a brief Google search to confirm.

File: cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml
Comment (knikolla) [reply cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml]: Makes sense. Are there any changes that we need to do on the Vault side of things to support this?

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml
Comment (joachimweyl) [reply cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: @knikolla is this change ready to be committed?

File: cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml
Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml]: There are a number of changes we will need to make in the vault configuration (although they are largely the same as we would need to make using a ClusterSecretStore).

**Comment by joachimweyl** [comment]
@knikolla as this is no longer blocked what are the next steps for this?

**Comment by larsks** [comment]
@joachimweyl it's still blocked until we've got the prod cluster minimally configured. There are some pull requests in flight for that right now, and then some work to get external secrets functioning correctly.

**Comment by joachimweyl** [comment]
@larsks I set this as blocked by [Add ocp-prod cluster-scope application to argocd](https://github.com/OCP-on-NERC/nerc-ocp-apps/pull/14) and that is in turn blocked by [Remove ocp-prod clusterversion](https://github.com/OCP-on-NERC/nerc-ocp-config/pull/96) are there any other PR's I should set it blocked by?

File: cluster-scope/base/config.openshift.io/oauths/cluster/oauth.yaml
Comment (larsks) [inline cluster-scope/base/config.openshift.io/oauths/cluster/oauth.yaml:11]: I don't think we need this change anymore; this file was updated in ee5f117 to have an empty `spec` which should be suffcient for patching.

File: cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml
Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-prod/externalsecrets/oauths-clientsecret-nerc.yaml]: This still needs to be updated:

```
spec:
  secretStoreRef:
    kind: SecretStore
    name: nerc-secret-store
```

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml
Comment (larsks) [reply cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: I'd like to see this updated to (a) revert the change to the ingresscontrollers patch and (b) use the same syntax for the `oauths/cluster_patch`.

**Review by larsks — CHANGES_REQUESTED** [review]
I think the dependencies are in place for this, but I think we still need a few minor changes to this PR before it can merge.

**Comment by joachimweyl** [comment]
@larsks since GitHub does not specifically alert you when your CR's are outdated I figured I would let you know that all of your CR's on this PR are now outdated.

**Review by larsks — APPROVED** [review]

**Pull request merged**
